### PR TITLE
Add support for passing request on render and retrieving in tag context

### PR DESCRIPTION
**Pull request by siemensikkema**

### CODE REVIEW

File: Sources/Sugar/Helpers/Render+userInfo.swift [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    /// - Throws: TagContextError.requestNotPassedIntoRender.
+    public func requireRequest() throws -> Request {
+        guard let request = request else {
+            throw TagContextError.requestNotPassedIntoRender
+        }
+        return request
+    }
+}
+
+private let requestUserInfoKey = "_submissions:request"
```
Comment (steffendsommer) [inline Sources/Sugar/Helpers/Render+userInfo.swift]: Should we name this more generic (and not include "submissions" here)?

**Review by steffendsommer — APPROVED** [review]
Consider renaming the key, otherwise LGTM

File: Sources/Sugar/Helpers/Render+userInfo.swift (hunk 1 above)
Comment (siemensikkema) [reply Sources/Sugar/Helpers/Render+userInfo.swift]: agreed, that was a copy/paste error. If you agree with the change please accept this suggestion:

```suggestion
private let requestUserInfoKey = "_sugar:request"
```

Comment (steffendsommer) [reply Sources/Sugar/Helpers/Render+userInfo.swift]: 👍

**Pull request merged**